index.js

```diff
@@ -17,6 +19,21 @@ module.exports = function (str, cb) {
 	});
 };
 
+module.exports.stream = function () {
+	var size = 0;
+	return zlib.createGzip(opts)
+		.on('data', function (buf) {
+			size += buf.length;
+		})
+		.on('error', function () {
+			this.size = 0;
+		})
+		.on('end', function () {
+			this.size = size;
+			this.emit('size', size);
+		});
+};
```

Comment:
Actually, I don't think it makes sense to return the gzip stream. The user only cares about the gzipped size. They don't actually want their stream gzipped. Rather return a passthrough stream with the event and property. I would also make the event name `gzip-size` and property `gzipSize` just to be explicit.